
///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2009, Arizona State University
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//  Author: Jeff Conner
//
///////////////////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////////////////
//
//  Input dialog for files.
//
///////////////////////////////////////////////////////////////////////////////

#include "AddChemicalDialog.h"

#include "Usul/Strings/Format.h"

///////////////////////////////////////////////////////////////////////////////
//
//  Constructor.
//
///////////////////////////////////////////////////////////////////////////////

AddChemicalDialog::AddChemicalDialog ( QWidget *parent ) : 
BaseClass ( parent ),
_chemical()
{
  //// Initialize code from Designer.
  this->setupUi ( this );  

  //initialize the table view
  this->_initialize();
}


///////////////////////////////////////////////////////////////////////////////
//
//  Destructor.
//
///////////////////////////////////////////////////////////////////////////////

AddChemicalDialog::~AddChemicalDialog()
{
  
}


///////////////////////////////////////////////////////////////////////////////
//
//  Initialize ( Public ).
//
///////////////////////////////////////////////////////////////////////////////

void AddChemicalDialog::initialize()
{
  this->_initialize();
}


///////////////////////////////////////////////////////////////////////////////
//
//  Initialize ( Private ).
//
///////////////////////////////////////////////////////////////////////////////

void AddChemicalDialog::_initialize()
{
 
}


///////////////////////////////////////////////////////////////////////////////
//
//  Create the chemical
//
///////////////////////////////////////////////////////////////////////////////

AddChemicalDialog::Chemical AddChemicalDialog::createChemical()
{
  // get the parameters from the text boxes
  std::string name        ( this->_name->text().toStdString()      );
  std::string hlaw        ( this->_hlaw->text().toStdString()      );
  std::string koc         ( this->_koc->text().toStdString()       );
  std::string diffAir     ( this->_diffAir->text().toStdString()   );
  std::string diffH2O     ( this->_diffWater->text().toStdString() );
  std::string atmoConc    ( this->_atmoConc->text().toStdString()  );

  // Create the chemical from user input
  Chemical c ( 0, name, hlaw, koc, diffAir, diffH2O, atmoConc );

  // set the chemical
  this->chemical ( c );

  // return the chemical
  return c;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the chemicals
//
///////////////////////////////////////////////////////////////////////////////

AddChemicalDialog::Chemical AddChemicalDialog::chemical()
{
  return _chemical;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Set the chemicals
//
///////////////////////////////////////////////////////////////////////////////

void AddChemicalDialog::chemical( Chemical c )
{
  _chemical = c;
}
